### ISSUE TITLE
[BUG - 3.0.0] Game crashing to desktop on closing the UE4SS console

### ISSUE THREAD

**Issue by hiKareeem** [UE4SS-RE/RE-UE4SS#336]
**Game and Engine Version**
Palworld 0.1.4.0
UE 5.1

**Describe the bug**
Game crashes to desktop on closing the UE4SS GUI console using either the x button or shift+O

**To Reproduce**
1. Launch game with GUI console enabled, or activate it with shift+O after launch
2. Close the GUI console with the x button or shift+O
3. Crash to desktop

**Expected behavior**
Expected GUI console to close while the game remains open as in 2.5.2

**Desktop (please complete the following information):**
 - OS: Win10
 
[UE4SS.log](https://github.com/UE4SS-RE/RE-UE4SS/files/14157678/UE4SS.log)
```
[2024-02-04 17:21:22] Console created
[2024-02-04 17:21:22] UE4SS - v3.0.0 Beta #0 - Git SHA #5319a0e
[2024-02-04 17:21:22] UE4SS Build Configuration: Game__Shipping__Win64 (MSVC)
[2024-02-04 17:21:22] Setting up mods...
[2024-02-04 17:21:22] Starting mods (from mods.txt load order)...
[2024-02-04 17:21:22] Starting mods (from enabled.txt, no defined load order)...
[2024-02-04 17:21:22] No specific game configuration found, using default configuration file
[2024-02-04 17:21:22] Config: C:\Games\Palworld\Pal\Binaries\Win64\UE4SS-settings.ini

[2024-02-04 17:21:22] root directory: C:\Games\Palworld\Pal\Binaries\Win64
[2024-02-04 17:21:22] working directory: C:\Games\Palworld\Pal\Binaries\Win64
[2024-02-04 17:21:22] game executable directory: C:\Games\Palworld\Pal\Binaries\Win64
[2024-02-04 17:21:22] game executable: C:\Games\Palworld\Pal\Binaries\Win64\Palworld-Win64-Shipping.exe (146885120 bytes)


[2024-02-04 17:21:22] mods directory: C:\Games\Palworld\Pal\Binaries\Win64\Mods
[2024-02-04 17:21:22] log directory: C:\Games\Palworld\Pal\Binaries\Win64
[2024-02-04 17:21:22] object dumper directory: C:\Games\Palworld\Pal\Binaries\Win64


[2024-02-04 17:21:22] PS Scan attempt 1
[2024-02-04 17:21:22] [PS] Reading image
[2024-02-04 17:21:22] [PS] Starting scan
[2024-02-04 17:21:23] [PS] Scan finished in 830.3634ms
[2024-02-04 17:21:23] [PS] Found EngineVersion: 5.1
[2024-02-04 17:21:23] [PS] Found GUObjectArray: 0x7ff7a5a5f6f0
[2024-02-04 17:21:23] [PS] Found GMalloc: 0x7ff7a59870a8
[2024-02-04 17:21:23] [PS] Found FName::ToString: 0x7ff79ff80d40
[2024-02-04 17:21:23] [PS] Found FName::FName(wchar_t*): 0x7ff79ff6fcd0
[2024-02-04 17:21:23] [PS] Found StaticConstructObject_Internal: 0x7ff7a0119b80
[2024-02-04 17:21:23] [PS] Found FText::FText(FString&&): 0x7ff79fea1580
[2024-02-04 17:21:23] PS scan successful
[2024-02-04 17:21:23] Already constructed: /Script/CoreUObject.Class
[2024-02-04 17:21:23] Already constructed: /Script/CoreUObject
[2024-02-04 17:21:23] Already constructed: /Script/CoreUObject.Struct
[2024-02-04 17:21:23] Already constructed: /Script/Engine.Pawn
[2024-02-04 17:21:23] Already constructed: /Script/Engine.Character
[2024-02-04 17:21:23] Already constructed: /Script/Engine.Actor
[2024-02-04 17:21:23] Need to construct: /Script/CoreUObject.Vector
[2024-02-04 17:21:23] Need to construct: /Script/Engine.Default__DefaultPawn
[2024-02-04 17:21:23] Need to construct: /Script/Engine.HitResult
[2024-02-04 17:21:23] Need to construct: /Script/Engine.Default__MaterialExpression
[2024-02-04 17:21:23] Already constructed: /Script/Engine.ActorComponent
[2024-02-04 17:21:23] Need to construct: /Script/CoreUObject.OrientedBox
[2024-02-04 17:21:23] Already constructed: /Script/Engine.MovementComponent
[2024-02-04 17:21:23] Already constructed: /Script/Engine.HUD
[2024-02-04 17:21:23] Already constructed: /Script/Engine.PlayerController
[2024-02-04 17:21:23] Already constructed: /Script/Engine.PlayerCameraManager
[2024-02-04 17:21:23] Need to construct: /Script/CoreUObject.EInterpCurveMode
[2024-02-04 17:21:23] Need to construct: /Script/Engine.ENetRole
[2024-02-04 17:21:23] Need to construct: /Script/MovieScene.MovieSceneEditorData
[2024-02-04 17:21:23] Already constructed: /Script/UMG.Widget
[2024-02-04 17:21:23] Already constructed: /Script/UMG.ComboBoxString
[2024-02-04 17:21:23] Already constructed: /Script/CoreUObject.Interface
[2024-02-04 17:21:23] Already constructed: /Script/CoreUObject.DynamicClass
[2024-02-04 17:21:24] Constructed [1 / 8]: /Script/Engine.HitResult
[2024-02-04 17:21:24] Constructed [2 / 8]: /Script/CoreUObject.OrientedBox
[2024-02-04 17:21:24] Constructed [3 / 8]: /Script/CoreUObject.EInterpCurveMode
[2024-02-04 17:21:24] Constructed [4 / 8]: /Script/Engine.ENetRole
[2024-02-04 17:21:24] Constructed [5 / 8]: /Script/CoreUObject.Vector
[2024-02-04 17:21:24] Constructed [6 / 8]: /Script/Engine.Default__DefaultPawn
[2024-02-04 17:21:24] Constructed [7 / 8]: /Script/Engine.Default__MaterialExpression
[2024-02-04 17:21:24] Constructed [8 / 8]: /Script/MovieScene.MovieSceneEditorData
[2024-02-04 17:21:25] Constructed 8 of 8 objects
[2024-02-04 17:21:26] ProcessInternal address: 0x7ff7a00f8590 <- Built-in
[2024-02-04 17:21:26] ProcessLocalScriptFunction address: 0x7ff7a00f8640 <- Built-in
```

**Comment by Buckminsterfullerene02** [comment]
Confirmed bug - pretty sure the cause is #322 due to it being the only commit that changes the GUI/live view between experimental builds 594 (that works) and 604 (that doesn't work). 

localcc will look into it when they have time to, thanks for the report!

**Comment by UE4SS** [comment]
> does this still happen since palworld 1.4.1

Fixed by #337, experimental build is currently compiling and should be ready within 30 minutes here: https://github.com/UE4SS-RE/RE-UE4SS/releases/tag/experimental

**Comment by UE4SS** [comment]
For anyone not wanting to wait for an official release: https://github.com/UE4SS-RE/RE-UE4SS/releases/download/experimental/zDEV-UE4SS_v3.0.0-1-g0352054.zip
